ggml/src/ggml-vulkan/vulkan-shaders/mul_mat_vec_q6_k.comp

```diff
@@ -4,9 +4,11 @@
 
 #include "mul_mat_vec_base.comp"
 
-layout(local_size_x = 32, local_size_y = 1, local_size_z = 1) in;
+layout(local_size_x_id = 0, local_size_y = 1, local_size_z = 1) in;
 
-shared FLOAT_TYPE tmp[32];
+layout (constant_id = 0) const uint SUBGROUP_SIZE = 32;
```

Comment:
I suggest to rename this to BLOCK_SIZE to be consistent with other shaders, and because this doesn't actually need to equal the subgroup size.

Reply:
If I'm being nitpicky I guess this should really be called WORKGROUP_SIZE as ggml already has an established definition for what a block is (a group of quantized values).

Reply:
If you're going to do that, I think it would be better to do it as a separate change, and change all instances of it at once. "Block" is the CUDA term for workgroup, so that's probably where the existing name came from.